test/jdk/java/awt/print/PrinterJob/SwingUIText.java

```diff
@@ -21,60 +21,145 @@
+import java.awt.GridLayout;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
 
 public class SwingUIText implements Printable {
 
-    static String[] instructions = {
-        "This tests that when a Swing UI is printed, that the text",
-        "in each component properly matches the length of the component",
-        "as seen on-screen, and that the spacing of the text is of",
-        "reasonable even-ness. This latter part is very subjective and",
-        "the comparison has to be with JDK1.5 GA, or JDK 1.6 GA",
-    };
-
-    static JFrame frame;
-
-    public static void main(String args[]) {
-        SwingUtilities.invokeLater(new Runnable() {
-          public void run() {
-              createUI();
-          }
-      });
+    private static JFrame frame;
+    private static JFrame testInstructionFrame;
+    private static final CountDownLatch resultCountDownLatch =
+            new CountDownLatch(1);
+    private static volatile boolean testResult = false;
+    private static volatile String failureReason;
+
+    public static void main(String[] args) throws InterruptedException, InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+            createTestInstructionUI();
+            createSwingTestUI();
+        });
+
+        // Giving max 10 minutes since sometime printer has to be ready to
+        // print and user has to compare the UI on screen and the print out.
+        if (!resultCountDownLatch.await(10, TimeUnit.MINUTES)) {
+            throw new RuntimeException("Timeout: User did not press either " +
+                    "Pass or Fail button");
+        }
+
+        if (!testResult) {
+            dispose();
+            throw new RuntimeException("Test Failed: " + failureReason);
+        } else {
+            System.out.println("Test Passed.");
+        }
     }
 
-    public static void createUI() {
+    private static void dispose() throws InterruptedException, InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+            if (frame != null) {
+                frame.dispose();
+            }
+            if (testInstructionFrame != null) {
+                testInstructionFrame.dispose();
+            }
+        });
+    }
 
-        Sysout.createDialogWithInstructions(instructions);
+    private static void createTestInstructionUI() {
+        testInstructionFrame = new JFrame("Test Instruction Frame");
+        final String INSTRUCTION = """
```

Comment:
Would this make more sense to use the already existing manual test support from the A11Y tests?